app/Models/User.php

```diff
@@ -374,6 +377,81 @@ public function getFilamentAvatarUrl(): ?string
         return $this->avatar_url ?: $this->getFirstTemporaryUrl(now()->addMinutes(5), 'avatar', 'avatar-height-250px');
     }
 
+    /**
+     * @param LicenseType | string | array<LicenseType | string> | null $type
+     */
+    public function hasLicense(LicenseType | string | array | null $type): bool
+    {
+        if (blank($type)) {
+            return true;
+        }
+
+        foreach (Arr::wrap($type) as $type) {
+            if (! ($type instanceof LicenseType)) {
+                $type = LicenseType::from($type);
+            }
+
+            if ($this->licenses->doesntContain('type', $type)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param LicenseType | string | array<LicenseType | string> | null $type
+     */
+    public function hasAnyLicense(LicenseType | string | array | null $type): bool
+    {
+        if (blank($type)) {
+            return true;
+        }
+
+        foreach (Arr::wrap($type) as $type) {
+            if (! ($type instanceof LicenseType)) {
+                $type = LicenseType::from($type);
+            }
+
+            if ($this->licenses->contains('type', $type)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param LicenseType | string | array<LicenseType | string> | null $type
+     */
+    public function scopeHasLicense(Builder $query, LicenseType | string | array | null $type): Builder
+    {
+        if (blank($type)) {
+            return $query;
+        }
+
+        foreach (Arr::wrap($type) as $type) {
+            $query->whereRelation('licenses', 'type', $type);
+        }
+
+        return $query;
+    }
+
+    /**
+     * @param LicenseType | string | array<LicenseType | string> | null $type
+     */
+    public function scopeHasAnyLicense(Builder $query, LicenseType | string | array | null $type): Builder
+    {
+        if (blank($type)) {
+            return $query;
+        }
+
+        return $query->whereHas(
+            'licenses',
+            fn (Builder $query) => $query->whereIn('type', Arr::wrap($type)),
+        );
+    }
```

Comment:
Same as the others, my preference would be for these to be invokable scopes.

Reply:
done